src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -62,27 +65,59 @@ public TimeSpan AutomaticRefreshInterval
         public static readonly TimeSpan DefaultAutomaticRefreshInterval = new TimeSpan(0, 12, 0, 0);
 
         /// <summary>
-        /// 5 minutes is the default time interval that must pass for <see cref="RequestRefresh"/> to obtain a new configuration.
+        /// 1 hour is the default time interval that a last known good configuration will last for.
         /// </summary>
-        public static readonly TimeSpan DefaultRefreshInterval = new TimeSpan(0, 0, 5, 0);
+        public static readonly TimeSpan DefaultLastKnownGoodConfigurationLifetime = new TimeSpan(0, 1, 0, 0);
 
         /// <summary>
-        /// The last known good configuration or LKG (a configuration retrieved in the past that we were able to successfully validate a token against).
+        /// 5 minutes is the default time interval that must pass for <see cref="RequestRefresh"/> to obtain a new configuration.
         /// </summary>
-        internal BaseConfiguration LastKnownGoodConfiguration { get; set; }
+        public static readonly TimeSpan DefaultRefreshInterval = new TimeSpan(0, 0, 5, 0);
 
         /// <summary>
         /// Obtains an updated version of <see cref="BaseConfiguration"/> if the appropriate refresh interval has passed.
         /// This method may return a cached version of the configuration.
         /// </summary>
         /// <param name="cancel">CancellationToken</param>
         /// <returns>Configuration of type Configuration.</returns>
-        internal abstract Task<BaseConfiguration> GetBaseConfigurationAsync(CancellationToken cancel);
+        public abstract Task<BaseConfiguration> GetBaseConfigurationAsync(CancellationToken cancel);
+
+        /// <summary>
+        /// The last known good configuration or LKG (a configuration retrieved in the past that we were able to successfully validate a token against).
+        /// </summary>
+        public BaseConfiguration LastKnownGoodConfiguration
+        {
+            get
+            {
+                // only set this value the first time the last known good configuration is used for validation
+                if (_lastKnownGoodConfigFirstUse.Equals(DateTime.MinValue)) _lastKnownGoodConfigFirstUse = DateTime.UtcNow;
```

Comment:
nit: consider reformatting this line

Reply:
Reformatted this to make it clearer by not trying to keep it all on the same line.
Also modified _lastKnownGoodConfigFirstUse to be a nullable Datetime, since DateTime.MinValue didn't feel like the right default for it.